src/components/DocumentProducer/EditImagesModal/PhotoSelector.tsx

```diff
@@ -142,14 +145,20 @@ export default function PhotoChooser(props: PhotoChooserProps): JSX.Element {
   }, [files, multipleSelection]);
 
   useEffect(() => {
+    if (!filesDataChanged) {
+      return;
+    }
+
     onPhotosChanged(filesData);
-  }, [filesData, onPhotosChanged]);
+    setFilesDataChanged(false);
+  }, [filesData, onPhotosChanged, filesDataChanged]);
```

Comment:
Changes were made in this file to prevent `onPhotosChanged` from being called until the field is modified and also prevent it from being called multiple times. Those changes seemed safe to make here, but I can clone this file and create a deliverable variation if preferred.